src/core/sync/mutex.d

```diff
@@ -144,7 +144,7 @@ class Mutex :
         {
             int rc = pthread_mutex_lock( &m_hndl );
             if( rc )
-                throw new SyncError( "Unable to lock mutex" );
+                throw cast(SyncError) cast(void*) typeid(SyncError).init;
```

Comment:
Did you want me to keep the error message in or not?(I was unsure if this was UB due to the init hack.)